models/flexload_milp.py

```diff
@@ -594,46 +617,57 @@ def solve_relaxedproblem(self, tee=False):
             ]
         )
         cont_load_avg = np.mean(cont_load)
+        lower_bound = (1 - flex_capacity) * cont_load_avg
+        upper_bound = (1 + flex_capacity) * cont_load_avg
 
-        # raise a warning if solution violates the bounds of the relaxed problem (within tolerance)
-        if (
-            np.max(np.abs(cont_load - cont_load_avg)) - self.tol
-            >= flex_capacity * cont_load_avg
-        ):
-            print(
-                "The solution violates the bounds of flexible operation."
-                "\nResolving the problem with increased bound penalties."
+        # print the continuous load bounds
+        print(
+            f"Continuous load bounds: [{lower_bound:.2f}, {upper_bound:.2f}] "
+            f"with average continuous load: {cont_load_avg:.2f}"
+        )
+        # print the continuous load
+        print(f"Max load: {np.max(cont_load):.2f}, Min load: {np.min(cont_load):.2f}")
+
+        resolve = 0
+
+        # check min continuous load bound
+        if lower_bound - self.model.min_contload.value > self.tol:
+            resolve = 1
+            self.model.min_contload_penalty.value = (
+                self.model.min_contload_penalty.value + descent_stepsize * (cont_load_avg - self.model.min_contload.value)
             )
-            # raise ValueError("Continuous load violated bounds of relaxed problem.")
-
-            # TODO - think about implementing a smarter algorithm for heuristic bound enforcement
-            stepsize = 1e-5
-
-            while (
-                np.max(np.abs(cont_load - cont_load_avg)) - self.tol
-                >= flex_capacity * cont_load_avg
-            ):
-                # get the violation of the maximum continuous load
-                max_contload_violation = np.max(
-                    [(1 + flex_capacity) * cont_load_avg - np.max(cont_load), 0]
-                )
-                self.model.max_contload_penalty.value = (
-                    self.model.max_contload_penalty.value
-                    + max_contload_violation * stepsize
-                )
+        else:
```

Comment:
I believe this `else` statement is unnecessary. By default if the `if` condition isn't met it will be skipped

Reply:
I should add that if your goal is to skip to the next iteration, then the next line should be `continue`. Currently it uses `pass` which is a noop that I only really use as a placeholder for future logic or to handle errors silently. Given the logic at the end of the `while` statement iterates the counter, I think that we don't want to use `continue`